### PR TITLE
feat(plugins): Front50UpdateRepository and required configuration

### PR DESCRIPTION
**Pull request by jonsie**
Closes https://github.com/spinnaker/spinnaker/issues/5241

Todo:

- [x] sanity check configuration
- [x] write some tests for the UpdateRepository

### CODE REVIEW

File: kork-plugins/src/main/java/com/netflix/spinnaker/config/PluginsAutoConfiguration.java [hunk 1]
```diff
@@ -83,23 +91,55 @@ public static SpinnakerPluginManager pluginManager(
...
-    Map<String, PluginRepositoryProperties> repositoriesConfig =
-        configResolver.resolve(
-            new RepositoryConfigCoordinates(),
-            new TypeReference<HashMap<String, PluginRepositoryProperties>>() {});
+      SpinnakerPluginManager pluginManager, List<UpdateRepository> updateRepositories) {
+    return new SpinnakerUpdateManager(pluginManager, updateRepositories);
+  }
+
+  @Bean
+  @ConditionalOnProperty("spinnaker.extensibility.repositories.front50.enabled")
```
Comment (robzienert) [inline kork-plugins/src/main/java/com/netflix/spinnaker/config/PluginsAutoConfiguration.java]: Move this to the other configuration class, too.

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/update/SpinnakerPluginInfo.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.kork.plugins.update
+
+import org.pf4j.update.PluginInfo
+
+data class SpinnakerPluginInfo(
```
Comment (robzienert) [inline kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/update/SpinnakerPluginInfo.kt]: We need the `requires` stuff on releases as well.

File: kork-plugins/src/main/java/com/netflix/spinnaker/config/Front50UpdateRepositoryConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+@Configuration
+@ConditionalOnProperty("spinnaker.extensibility.repositories.front50.enabled")
```
[the remark this answers is not in the document]
Comment (jonsie) [reply kork-plugins/src/main/java/com/netflix/spinnaker/config/Front50UpdateRepositoryConfiguration.java]: `matchIfMissing` defaults to false, so I don't think I need to change anything here.

File: kork-plugins/src/main/java/com/netflix/spinnaker/config/Front50UpdateRepositoryConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+public class Front50UpdateRepositoryConfiguration {
+
```
[the remark this answers is not in the document]
Comment (jonsie) [reply kork-plugins/src/main/java/com/netflix/spinnaker/config/Front50UpdateRepositoryConfiguration.java]: I removed all those unnecessary beans - nice call.  In the future we could move those out to separate beans if we have more remote UpdateRepositories we want to support, but for now I like only creating what we need.

**Pull request closed without merging**